amaranth-chrome-ext/src/AmaranthUtil.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Utility class with helpful functions for the Amaranth Chrome extension.
+ * All methods are static.
+ */
+class AmaranthUtil {
+  /**
+   * Removes special characters from a string.
+   * @param {string} str String to remove special characters from
+   * @param {string} filters Characters to remove from str
+   * @return {string} `str` with all characters in `filters` removed
+   */
+  static removeSpecialCharacters(
+      str, filters='!"#$%&()*+,-./:;<=>?@[\\]^_`{|}~\t\n') {
+    let newStr = '';
+    for (const char of str) {
+      if (!filters.includes(char)) {
+        newStr += char;
+      }
+    }
+
+    return newStr;
+  }
+
+  /**
+   * Pads some array `arr` with `padValue` until it's length is equal to
+   * `desiredLength`. This function will modify the original array `arr` and
+   * return it. If the length of `arr` is already greater than `desiredLength`,
+   * the original `arr` is returned unchanged.
+   * @template T
+   * @param {T[]} arr The array to pad
+   * @param {number} desiredLength The length that `arr` should be after padding
+   * @param {T} padValue The value to add to `arr` to reach length
+   * `desiredLength`
+   * @return {T[]} The same array `arr` with as many `padValue`s added as
```

Comment:
Is this actually returning the "same" array? Is it the original array with more space allocated? Is the original array modified in place? Or did you just make a copy of the array and add padding values.

Reply:
The original array is modified in-place. JS is pass-by-value, but arrays are references, so the original is modified when I do arr.push()

Reply:
Will also duplicate `length` > `desiredLength` clause in `@return` section

Reply:
Fixed in https://github.com/googleinterns/amaranth/pull/44/commits/191cef03fbe2a27aa86278956438c4068992bd11